lib/csv_export.rb

```diff
@@ -0,0 +1,70 @@
+module CsvExport
+  require 'csv'
+  require 'zip'
+
+  # generates a csv from the given model data
+  def generate_csv(data)
+    objects = data.first
+    columns = data.size == 1 ? objects.first.attributes.keys : data[1]
+
+    # we never need to export ids
+    columns.delete('id')
+
+    CSV.generate(headers: true) do |csv|
+      csv << columns
+
+      objects.each do |o|
+        csv << columns.map do |attr|
+          s = o.send(attr)
+          if s.is_a? ActiveRecord::Base
+            s.name
+          else s
+          end
+        end
+      end
+    end
+  end
+
+  # generates a zip file containing multiple CSVs
+  def generate_zip(data)
```

Comment:
This is ok for me, it's clearer from `generate_csv` (and your new comments) what's going on.